engine/src/main/java/org/hibernate/search/spi/SearchIntegratorBuilder.java

```diff
@@ -499,37 +497,6 @@ private static IndexingMode defineIndexingMode(SearchConfiguration cfg) {
 		return IndexingMode.fromExternalRepresentation( indexingStrategy );
 	}
 
-	private ErrorHandler createErrorHandler(SearchConfiguration searchConfiguration) {
-		Object configuredErrorHandler = searchConfiguration.getProperties().get( Environment.ERROR_HANDLER );
```

Comment:
Let me detail my use case:

```
class MyErrorHandler {

  public MyErrorHandler(ErrorHandler configuredErrorHandler) {
      ...
  }
  ...
}
```

By moving the creation to the Configuration level, it allows me to decorate the configured ErrorHandler and return it.

Not sure about Services: would it create a single ErrorHandler per JVM? One ErrorHandler per ```SearchIntegrator``` is enough.

What about undoing everything and exposing the creation logic at SPI level ```org.hibernate.search.spi.ErrorHandlerCreator``` instead? I can simply re-use it as a building block for my custom ErrorHandler